tests/phpunit/tests/post/getPages.php

```diff
@@ -325,6 +325,141 @@ public function test_get_pages_include_exclude() {
 		$this->assertSame( $inc, $exc_result );
 	}
 
+	/**
+	 * @ticket 12821
+	 */
+	public function test_get_pages_test_filter() {
+		register_post_type( 'wptests_pt', array( 'hierarchical' => true ) );
+
+		$posts             = self::factory()->post->create_many(
+			2,
+			array(
+				'post_type' => 'wptests_pt',
+			)
+		);
+		$query_args_values = array();
+
+		// Filter the query to return the wptests_pt post type.
+		add_filter(
+			'get_pages_query_args',
+			static function( $query_args ) use ( &$query_args_values ) {
+				$query_args['post_type'] = 'wptests_pt';
+				$query_args_values       = $query_args;
+				return $query_args;
+			}
+		);
```

Comment:
Since two args are passed to the filter, I recommend also capturing the `$parsed_args` and making sure it was passed, and has the expected shape below as well.

Reply:
What is the benefit of that?

Reply:
To ensure the expected argument is passed. We have phpdoc for a second parameter, but that second parameter isn't being tested anywhere. So if the parameter were to be removed or changed in some way, plugins that rely on it would break.

Reply:
See suggestion below: https://github.com/WordPress/wordpress-develop/pull/4386/files#r1199343781

Reply:
We don't have any tests like this in core. At this point we are just testing the filter system works. Testing that filter can change the value makes sense, I dont agree with the need to test for this argument.

Reply:
I disagree. And actually we do have this in core. For example:

https://github.com/WordPress/wordpress-develop/blob/faee35ab0d872b4b2d53b6d50664e68588b84a78/tests/phpunit/tests/customize/manager.php#L2631

https://github.com/WordPress/wordpress-develop/blob/faee35ab0d872b4b2d53b6d50664e68588b84a78/tests/phpunit/tests/customize/manager.php#L2673-L2689

Reply:
@westonruter Fair enough. I have now added tests for the second parameter as well. We have lots of asserts now!

Reply:
Thank you!